modules/social_features/social_topic/social_topic.module

```diff
@@ -39,10 +40,12 @@ function social_topic_preprocess_node(array &$variables) {
     $topic_type = $node->get('field_topic_type');
     $topic_type_entities = $topic_type->referencedEntities();
     if (count($topic_type_entities) === 1) {
+      $curr_langcode = \Drupal::languageManager()->getCurrentLanguage(LanguageInterface::TYPE_CONTENT)->getId();
       foreach ($topic_type_entities as $topic) {
         $variables['metadata'] = t('in @topic', ['@topic' => $topic->link()]);
         // Set topic type link.
         $topic_type_url = Url::fromRoute('view.latest_topics.page_latest_topics', ['field_topic_type_target_id' => $topic->id()]);
+        $topic->isTranslatable() ? $topic = $topic->getTranslation($curr_langcode) : NULL;
         $topic_type_link = Link::fromTextAndUrl($topic->label(), $topic_type_url)
           ->toString();
         $variables['topic_type'] = $topic_type_link;
```

Comment:
I think we can (and should) use this PR to replace this line with

```suggestion
         $variables['topic_type'] = $topic->toLink();
```

I've chosen for `toLink` because it creates a render array that can be modified instead of a string. If that doesn't work then we can use `link()` instead which will output a string.

Reply:
@Kingdutch your suggested modification of line 53 will result in a link to:
/taxonomy/term/[termID]
instead of
/all-topics?field_topic_type_target_id=[termID]
which is not correct